catroid/src/org/catrobat/catroid/physics/PhysicsObject.java

```diff
@@ -333,6 +328,19 @@ protected void setCollisionBits(short categoryBits, short maskBits) {
 			filter.maskBits = maskBits;
 			fixture.setFilterData(filter);
 		}
+
+		if (updateState) {
+			updateNonCollidingState();
+		}
+	}
+
+	private void updateNonCollidingState() {
+		if (body.getUserData() != null && body.getUserData() instanceof Sprite) {
+			Object look = ((Sprite) body.getUserData()).look;
+			if (look != null && look instanceof PhysicsLook) {
+				((PhysicsLook) look).setNonColliding(collisionMaskRecord == PhysicsWorld.MASK_NOCOLLISION);
+			}
```

Comment:
As far as I can tell, the body knows the sprite only via the "userData", which is what I am using at the moment. Is there something I'm missing here?

I'm sorry but I can't think of another/a better way to access the object's respective physicsLook - at least not without changing/adding significant amount of code - perhaps adding a direct reference but I think that's out of the question.